azure/README.md

```diff
@@ -0,0 +1,104 @@
+# Pytorch Elastic on Azure
+This directory contains scripts and libraries that help users run pytorch elastic jobs on Azure.
+
+## Prerequisites
+1. Familiarity with [Azure](https://azure.microsoft.com/en-us/), [aks-engine](https://github.com/Azure/aks-engine), [Azure Blob Storage](https://azure.microsoft.com/en-us/services/storage/blobs/)
+2. Quota available for Standard_DS1_v2 instance and Standard_NC6s_v3 instance.
+3. Access to Azure subscription, Resource Group and Storage. (Refer [here](https://github.com/Azure/MachineLearningNotebooks/blob/master/configuration.ipynb) for setup instructions).
+
+# Sample Usage
+
+1. #### Configure your job yaml and kubernetes.json
+```
+python petctl.py configure --name "test_job" --min_size 1 --max_size 5
+```
+This will create a spec for aks-engine instances and training job. Aks-engine launch [spec](config/kubernetes.json) is a simple json file that specifies the count and type of Rendezvous and Worker instances. A training job [spec](config/sample_specs.yaml) file is created with the specified job name and min, max worker count.
+By default master node creates Standard DS1_v2 instance and worker nodes create Standard_NC6_v3 instances. Other Azure instances could be specified using --master_vm and --worker_vm options.
+
+2. #### Setup your Kubernetes cluster
+
+This step requires service prinicpal to create aks cluster.  
+Instructions for generating service principal can be found at [portal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal), [CLI](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest), [Powershell](https://docs.microsoft.com/en-us/powershell/azure/create-azure-service-principal-azureps).  
+```
+python petctl.py setup --dns_prefix azure-pytorch-elastic 
+                       --rg "<resource_group>" 
+                       --location "<location>" 
+                       --subscription_id <subscription_id>
+                       --client_id <service principal client id>
+                       --client_secret <service principal client secret>                       
+```
+This creates an Azure Kubernetes cluster with 1 Standard_DS1_v2 master instances and specified number of Standard_NC6s_v3 worker instance in the resource group created in [Prerequisites](#Prerequisites) #3.
+
+3. #### Upload to Azure Blob storage
+
+This is an optional step to upload code and data to Azure blob storage. This step can be skipped if the training script and data are already available in Azure Blob storage.
+```
+python petctl.py upload_storage --source_path <path to files>
```

Comment:
Rather than building this functionality into `petctl`, you could suggest using `azcopy` instead?

Reply:
This one is similar to AWS upload_script petctl functionality, it uploads the training scripts, etc. to Azure Blob storage.